lib/waffle/file.ex

```diff
@@ -158,19 +158,31 @@ defmodule Waffle.File do
 
     case remote_file do
       {:ok, body, filename} ->
-        case File.write(local_path, body) do
+        case write_to_file(local_path, body) do
           :ok -> {:ok, filename}
           _ -> :error
         end
 
       {:ok, body} ->
-        File.write(local_path, body)
+        :ok = write_to_file(local_path, body)
+
+
 
       {:error, _reason} = err ->
         err
     end
   end
 
+  #stream binary in chuck into the file. Fixes the write for >2GB files.
+  defp write_to_file(local_path, body) do
+    body
+    |> StringIO.open()
+    |> elem(1)
+    |> IO.binstream(1024*1024*10)
```

Comment:
why this number exactly?

Reply:
It arbitrary. I'm using 10MB chunks to write to files. When to small it can be long when to high it will use to much memory i figured 10Mb is good compromise today. I don't know what would be best here :)